### PR TITLE
Exclude Kubernetes-specific variables

### PR DESCRIPTION
**Pull request by alvarosanchez**
Fixes https://github.com/micronaut-projects/micronaut-kubernetes/issues/209

### CODE REVIEW

**Comment by jameskleeh** [comment]
@alvarosanchez It seems likely to me that some users may be relying on this data, and thus it should be opt in somehow. Do you think it's unreasonable to believe some folks are reading this data from config?

File: inject/src/main/java/io/micronaut/context/env/KubernetesEnvironmentPropertySource.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    && !entry.getKey().endsWith("_TCP_PORT")
+                    && !entry.getKey().endsWith("_TCP_PROTO")
+                    && !entry.getKey().endsWith("_TCP_ADDR")
+                    && !entry.getKey().endsWith("_UDP_PORT")
+                    && !entry.getKey().endsWith("_UDP_PROTO")
+                    && !entry.getKey().endsWith("_UDP_ADDR")
+                    && !entry.getKey().endsWith("_SERVICE_PORT")
+                    && !entry.getKey().endsWith("_SERVICE_PORT_HTTP")
+                    && !entry.getKey().endsWith("_SERVICE_HOST")
+                    && (!entry.getKey().endsWith("_PORT") && !entry.getValue().startsWith("tcp://"))
```
Comment (jameskleeh) [inline inject/src/main/java/io/micronaut/context/env/KubernetesEnvironmentPropertySource.java]: Some of these checks are redundant. `entry.getKey().endsWith("_PORT")` should cover `_TCP_PORT`, `_UDP_PORT`, `_SERVICE_PORT`

Comment (jameskleeh) [reply inject/src/main/java/io/micronaut/context/env/KubernetesEnvironmentPropertySource.java]: Perhaps these string suffixes should be put in a list and you can just loop

**Review by jameskleeh — CHANGES_REQUESTED** [review]

File: inject/src/main/java/io/micronaut/context/env/KubernetesEnvironmentPropertySource.java (hunk 1 above)
Comment (alvarosanchez) [reply inject/src/main/java/io/micronaut/context/env/KubernetesEnvironmentPropertySource.java]: Yeah I copied the code from the original poster of the issue. Will refactor it a bit

**Comment by alvarosanchez** [comment]
@jameskleeh the old `KubernetesDiscoveryClient`, which relied on those environment variables, was deprecated in Micronaut 1.x and removed in 2.x. The Micronaut Kubernetes modules doesn't use environment variables.

I don't see a reason why anyone would access those variables directly, but in any case, they will always be available via `System.getenv()`.

**Comment by graemerocher** [comment]
yeah typically you would use the Kubernetes module which would use those variables so I don't think there is any reason not to merge this change

**Pull request merged**